lib/bugcrowd_templates/template_path.rb

```diff
@@ -17,18 +17,54 @@ def initialize(type:, field:, category: '', subcategory: '', item: '', file_name
     end
     # rubocop:enable Metrics/ParameterLists, Metrics/CyclomaticComplexity
 
-    def template_file(type)
+    def template_file
       validate_input_attrs
 
-      file_pathname = case type
-                      when 'submissions'
-                        BugcrowdTemplates.current_directory.join(
-                          type, field, category, subcategory, item, file_name
-                        )
-                      when 'methodology'
-                        BugcrowdTemplates.current_directory.join(type, field, category, subcategory, item)
-                      end
-      "#{file_pathname}.md"
+      find_template_file
+    end
+
+    # find the template file from  `category/subcategory/item` directories, checking from item to category level
+    # when dir is `item` then it will return `.../submissions/description/category/subcategory/item/template.md`
+    # when dir is `subcategory` then it will return  `.../submissions/description/category/subcategory/template.md`
+    # when dir is `category` then it will return `.../submissions/description/category/template.md`
+    def find_template_file
+      return item_file_path if item && File.exist?(item_file_path)
+      return subcategory_file_path if subcategory && File.exist?(subcategory_file_path)
+      category_file_path
+    end
+
+    # check upto item directory, and return if template exist
+    # e.g: `templates/submissions/description/server_security_misconfiguration/clickjacking/form_input/template.md`
+    def item_file_path
+      item_path = BugcrowdTemplates.current_directory.join(
+        type, field, category, subcategory, item, file_name
+      )
+
+      file_path_url(item_path)
+    end
+
+    # check upto subcategory directory, and return if template exist
+    # e.g: `templates/submissions/description/server_security_misconfiguration/clickjacking/template.md`
+    def subcategory_file_path
+      subcategory_path = BugcrowdTemplates.current_directory.join(
+        type, field, category, subcategory, file_name
+      )
```

Comment:
the `file_path_url(file_path)` method looks confusing to me.
what about building everything inside these methods – even with some repetition ?

```suggestion
      file_name_with_extension = [file_name, 'md'].join('.')
      
      subcategory_path = BugcrowdTemplates.curmrent_directory.join(
        type, field, category, subcategory, file_name_with_extension
      )
```

Reply:
I agree your points, but it will not work for methodologies, since the methodologies we don't have `template` file separately. so we will pass the `file_name` as empty. in that case it will give the wrong url path for methodologies

1. for submissions it will work perfectly, file_name = 'template'
when file_name_with_extension = [file_name, 'md'].join('.') => `template.md`

```
 subcategory_path = BugcrowdTemplates.current_directory.join(
  'submissions', 'description', 'category', 'subcategory', 'template.md'
)
=> #<Pathname:/Users/rajkumarulaganathan/Projects/templates/submissions/description/category/subcategory/template.md>
```
2. for methodologies it will not work, file_name is ''
when file_name_with_extension = [file_name, 'md'].join('.') => `.md`

```
subcategory_path = BugcrowdTemplates.current_directory.join(
  'methodologies', 'notes', 'website_testing', 'information', '.md'
)
=> #<Pathname:/Users/rajkumarulaganathan/Projects/templates/methodologies/notes/website_testing/information/.md>
```

The original file path is : `<Pathname:/Users/rajkumarulaganathan/Projects/templates/methodologies/notes/website_testing/information.md>`

the problem with `information/.md` instead of `information.md` for methodologies

The other solution is that, if we have `template.md` instead of `information.md` in our methodologies then it is good to go with suggested method `file_name_with_extension`. any thoughts?

Reply:
can't we just pass a file name for methodologies as well?

Reply:
yes we can then what will the file_name for that?..  the file_name will be `information` instead of `template`?. 
e.g: category: `website_testing` and file_name: `information` is that fine?

Reply:
yeah why not

Reply:
we make it consistent and explicit at the same time

Reply:
sure i will do the change

Reply:
Updated the changes